### PR TITLE
Fix thirdparty parameter

### LINKED ISSUE #21919: Watch of 3rd party objects needs a test
A user report that it's broken in the slack channel: https://gist.github.com/cgilmour/4c8e721691f61e711fa0

```
E0224 21:54:28.145127   24219 errors.go:63] apiserver received an error that is not an unversioned.Status: no kind "ListOptions" is registered for version "romana.io/demo/v1"
```

### PR DESCRIPTION
**Pull request by caesarxuchao**
Partially fix #21919.
#21919 reports query parameter (e.g., ListOptions) is not properly decoded by a resthandler of thirdparty resource. The error occurs because schema cannot create a thirdparty option object. Because we don't yet have a way to register a thirdparty option to the schema, I just add a wrapper to the codec, so that the wrapped codec always assumes the query parameter is converted from a v1 option.

This PR has a simple unit test. I prefer adding e2e tests for thirdparty resources in another PR.

cc @brendandburns @smarterclayton @lavalamp

### CODE REVIEW

**Comment by smarterclayton** [comment]
LGTM

File: pkg/master/master_test.go [hunk 1]
```diff
@@ -829,6 +829,40 @@ func httpDelete(url string) (*http.Response, error) {
...
+	_, etcdserver, server, assert := initThirdParty(t, version)
+	// TODO: Uncomment when fix #19254
+	// defer server.Close()
+	defer etcdserver.Terminate(t)
+
+	// send a GET request with query parameter
+	resp, err := httpGetWithQuery(server.URL + "/apis/company.com/" + version + "/namespaces/default/foos")
+	if !assert.NoError(err) {
+		t.Errorf("unexpected error: %v", err)
+		return
```
Comment (lavalamp) [inline pkg/master/master_test.go]: Nit: use Failf and then you don't need the return

File: pkg/master/master_test.go [hunk 2]
```diff
@@ -829,6 +829,40 @@ func httpDelete(url string) (*http.Response, error) {
...
+	}
+	assert.Equal(http.StatusOK, resp.StatusCode)
+}
+
+func httpGetWithQuery(url string) (*http.Response, error) {
```
Comment (lavalamp) [inline pkg/master/master_test.go]: nit: httpGetWithRV?

**Comment by lavalamp** [comment]
LGTM, fix nits only if you see this before it merges :)

**Comment by caesarxuchao** [comment]
Nits fixed.

**Comment by bgrant0607** [comment]
@k8s-bot test this issue: #21484

**Pull request merged**